### PR TITLE
etesync-dav: 0.20.0 -> 0.30.6

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

etesync-dav was stuck on 0.20.0 due to higher versions depending on python bindings to etebase. This pull request updates etesync-dev and includes the nix expression for etebase.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/etesync-dav/default.nix [hunk 1]
```diff
@@ -2,31 +2,27 @@
...
   src = python3Packages.fetchPypi {
     inherit pname version;
-    sha256 = "1q8h89hqi4kxphn1g5nbcia0haz5k57is9rycwaabm55mj9s9fah";
+    sha256 = "0cjz4p3a750fwvrxbzwda0sidw7nscahvppdshbsx49i6qrczpbg";
   };
 
   postPatch = ''
-    substituteInPlace setup.py --replace "Radicale==" "Radicale>="
+    substituteInPlace setup.py
   '';
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/etesync-dav/default.nix:14]: Remove this entire block if it is not needed.

File: pkgs/applications/misc/etesync-dav/default.nix [hunk 2]
```diff
@@ -2,31 +2,27 @@
...
   ];
 
-  checkInputs = with python3Packages; [
-    pytest
-  ];
-
-  checkPhase = ''
-    pytest
-  '';
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/etesync-dav/default.nix:24]: Either activate doCheck or do an pythonImportsCheck.

File: pkgs/applications/misc/etesync-dav/default.nix [hunk 3]
```diff
@@ -2,31 +2,27 @@
...
+  doInstallCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/etesync-dav/default.nix:25]:
```suggestion
```

File: pkgs/development/python-modules/etebase/default.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "etebase";
+  version = "0.31.1";
+
+  src = fetchFromGitHub {
+    owner = "etesync";
+    repo = "etebase-py";
+    rev = "950902a520417eaaf34d2c5e5e590ddb3c90358e";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/etebase/default.nix:22]: Please use an upstream tag if available.

File: pkgs/development/python-modules/etebase/default.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    pipInstallPhase
+  '';
+
+  meta = with lib; {
+    homepage = "https://www.etebase.com/";
+    description = "A Python client library for Etebase";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ _3699n ];
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/etebase/default.nix:61]:
```suggestion
```

File: pkgs/development/python-modules/etebase/default.nix [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  version = "0.31.1";
+
+  src = fetchFromGitHub {
+    owner = "etesync";
+    repo = "etebase-py";
+    rev = "950902a520417eaaf34d2c5e5e590ddb3c90358e";
+    sha256 = "163iw64l8lwawf84qswcjsq9p8qddv9ysjrr3dzqpqxb2yb0sy39";
+    fetchSubmodules = true;
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/etebase/default.nix:26]:
```suggestion
```

File: pkgs/development/python-modules/etebase/default.nix [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+rustPlatform.buildRustPackage rec {
+  pname = "etebase";
+  version = "0.31.1";
+
+  src = fetchFromGitHub {
+    owner = "etesync";
+    repo = "etebase-py";
+    rev = "950902a520417eaaf34d2c5e5e590ddb3c90358e";
+    sha256 = "163iw64l8lwawf84qswcjsq9p8qddv9ysjrr3dzqpqxb2yb0sy39";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/etebase/default.nix:24]: The repo is not using any submodules so this is not needed.

File: pkgs/development/python-modules/etebase/default.nix [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+  };
+
+
+  cargoSha256 = "0w8ypl6kj1mf6ahbdiwbd4jw6ldxdaig47zwk91jjsww5lbyx4lf";
+
+  nativeBuildInputs = [
+    rustfmt
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/etebase/default.nix:31]: Why do we need to format the code?

File: pkgs/development/python-modules/etebase/default.nix [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    setuptools-rust
+    wheel
+  ];
+
+  propagatedBuildInputs = [
+    python
+    msgpack
+  ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/etebase/default.nix:44]: Either activate doCheck or do an pythonImportsCheck.

**Review by SuperSandro2000 — COMMENTED** [review]
Also squash the maintainers adding commit with the update.

File: pkgs/development/python-modules/etebase/default.nix (hunk 8 above)
Comment (3699n) [reply pkgs/development/python-modules/etebase/default.nix:31]: It's a dependency of flapigen and therefore also required for etebase.

File: pkgs/applications/misc/etesync-dav/default.nix (hunk 2 above)
Comment (3699n) [reply pkgs/applications/misc/etesync-dav/default.nix:24]: 0.30.6 doesn't come with any tests and `pythonImportsCheck = [ "etesync-dav" ]` fails. What do you recommend the best solution is in this case?

Comment (3699n) [reply pkgs/applications/misc/etesync-dav/default.nix:24]: nixpkgs also doesn't yet package etebase/etesync, so I am unsure how you would write a test.

Comment (3699n) [reply pkgs/applications/misc/etesync-dav/default.nix:20]: I wrote a nix test myself, but I still left `doCheck = false`. Is this fine?

Comment (SuperSandro2000) [reply pkgs/applications/misc/etesync-dav/default.nix:20]: > 0.30.6 doesn't come with any tests

Maybe they do not ship them on pypi. You could try the official tarball release.

> and `pythonImportsCheck = [ "etesync-dav" ]` fails.

Oh that can happen with applications which are not supposed to be imported. I thought it was a module.

> What do you recommend the best solution is in this case?

Just disable the tests or try the source repo tarball.

> I wrote a nix test myself, but I still left doCheck = false. Is this fine?

Yeah, this is fine.

File: pkgs/development/python-modules/etebase/default.nix [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    wheel
+  ];
+
+  propagatedBuildInputs = [
+    python
+    msgpack
+  ];
+
+  doCheck = true;
+  #doInstallCheck = true;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/etebase/default.nix:43]:
```suggestion
```

File: pkgs/development/python-modules/etebase/default.nix [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    pipInstallPhase
+  '';
+
+  meta = with lib; {
+    homepage = "https://www.etebase.com/";
+    description = "A Python client library for Etebase";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ _3699n ];
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/etebase/default.nix:59]: You left the extra whitespace in.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105519` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>etesync-dav</li>
    <li>python37Packages.etebase</li>
    <li>python38Packages.etebase</li>
    <li>python39Packages.etebase</li>
  </ul>
</details>

**Comment by 3699n** [comment]
Okay I made another push, which should have fixed any remaining issues.

File: pkgs/applications/networking/browsers/ephemeral/default.nix [hunk 12]
```diff
@@ -20,13 +20,13 @@
 
 stdenv.mkDerivation rec {
   pname = "ephemeral";
-  version = "6.4.1";
+  version = "7.0.4";
```
Comment (Valodim) [inline pkgs/applications/networking/browsers/ephemeral/default.nix:23]: seems like this unrelated commit snuck in?

**Review by Valodim — COMMENTED** [review]
lgtm :+1: thanks for putting in the work

File: pkgs/applications/networking/browsers/ephemeral/default.nix (hunk 12 above)
Comment (3699n) [reply pkgs/applications/networking/browsers/ephemeral/default.nix]: Thanks. Just removed it.

**Review by Valodim — APPROVED** [review]
lgtm now. you might wanna add yourself to the maintainer list of etesync-dav too, so you get notifications on related PRs for that

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105519` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>etesync-dav</li>
    <li>python37Packages.etebase</li>
    <li>python38Packages.etebase</li>
    <li>python39Packages.etebase</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
patch to compile etebase on darwin:

```patch
diff --git a/pkgs/development/python-modules/etebase/default.nix b/pkgs/development/python-modules/etebase/default.nix
index c487aa98eb17..366c447ed942 100644
--- a/pkgs/development/python-modules/etebase/default.nix
+++ b/pkgs/development/python-modules/etebase/default.nix
@@ -1,4 +1,4 @@
-{ lib
+{ stdenv
 , wheel
 , rustPlatform
 , pipInstallHook
@@ -10,6 +10,7 @@
 , perl
 , rustfmt
 , fetchFromGitHub
+, Security
 }:
 
 rustPlatform.buildRustPackage rec {
@@ -34,6 +35,8 @@ rustPlatform.buildRustPackage rec {
     wheel
   ];
 
+  buildInputs = stdenv.lib.optionals stdenv.isDarwin [ Security ];
+
   propagatedBuildInputs = [
     python
     msgpack
@@ -49,7 +52,7 @@ rustPlatform.buildRustPackage rec {
     pipInstallPhase
   '';
 
-  meta = with lib; {
+  meta = with stdenv.lib; {
     homepage = "https://www.etebase.com/";
     description = "A Python client library for Etebase";
     license = licenses.bsd3;
diff --git a/pkgs/top-level/python-packages.nix b/pkgs/top-level/python-packages.nix
index 10328cb08e8c..5010c7dcc8be 100644
--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -1951,7 +1951,9 @@ in {
 
   etelemetry = callPackage ../development/python-modules/etelemetry { };
 
-  etebase = callPackage ../development/python-modules/etebase { };
+  etebase = callPackage ../development/python-modules/etebase {
+    inherit (pkgs.darwin.apple_sdk.frameworks) Security;
+  };
 
   etesync = callPackage ../development/python-modules/etesync { };
```

File: maintainers/maintainer-list.nix [hunk 13]
```diff
@@ -64,6 +64,12 @@
     githubId = 791309;
     name = "Jan Hrnko";
   };
+  _3699n = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:67]: Please put this into a separate commit with the message ``maintainers: add _3699n``.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 13 above)
Comment (3699n) [reply maintainers/maintainer-list.nix:67]: Okay I can do that. I thought you asked me to squash it?

Comment (SuperSandro2000) [reply maintainers/maintainer-list.nix:67]: I meant to squash review changes into the update commit.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105519` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>etesync-dav</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.etebase</li>
    <li>python38Packages.etebase</li>
    <li>python39Packages.etebase</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105519` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>etesync-dav</li>
    <li>python37Packages.etebase</li>
    <li>python38Packages.etebase</li>
    <li>python39Packages.etebase</li>
  </ul>
</details>
